src/cpp/fastdds/xtypes/type_representation/TypeObjectUtils.cpp

```diff
@@ -2050,6 +2050,51 @@ void TypeObjectUtils::plain_collection_type_identifier_header_consistency(
     }
 }
 
+void TypeObjectUtils::plain_map_type_identifier_header_consistency(
+        const PlainCollectionHeader& header,
+        const TypeIdentifier& element_identifier,
+        const TypeIdentifier& key_identifier)
+{
+    switch (header.equiv_kind()){
+        case EK_BOTH:
+            if (!is_fully_descriptive_type_identifier(key_identifier))
+            {
+                throw InvalidArgumentError(
+                          "The given header equiv_kind is EK_BOTH, but the key_identifier is not fully descriptive.");
+            }
+            if (!is_fully_descriptive_type_identifier(element_identifier))
+            {
+                throw InvalidArgumentError(
+                          "The given header equiv_kind is EK_BOTH, but the element_identifier is not fully descriptive.");
+            }
+            break;
+        case EK_MINIMAL:
+            if (!(is_fully_descriptive_type_identifier(key_identifier) || key_identifier._d() == EK_MINIMAL))
+            {
+                throw InvalidArgumentError(
+                          "The given header equiv_kind is EK_MINIMAL, but the key_identifier is neither fully descriptive nor EK_MINIMAL.");
+            }
+            if (!(is_fully_descriptive_type_identifier(element_identifier) || element_identifier._d() == EK_MINIMAL))
+            {
+                throw InvalidArgumentError(
+                          "The given header equiv_kind is EK_MINIMAL, but the element_identifier is neither fully descriptive nor EK_MINIMAL.");
+            }
```

Comment:
On the one hand, I do not think this is checking every possible case. It could be that the equivalence kind is EK_MINIMAL and with the current implementation, both a fully descriptive element and key TypeIdentifier will pass the check, when that is not correct because in that case the equivalence kind should be EK_BOTH.

This suggestion and the next one also applies to the COMPLETE case, below.

Reply:
changed in [7e1e448](https://github.com/eProsima/Fast-DDS/pull/4496/commits/7e1e4489afe7d44ecfa0a0767b420b5eaf5d28eb)